src/client/stdlib/duckdb.js

```diff
@@ -182,6 +183,22 @@ Object.defineProperty(DuckDBClient.prototype, "dialect", {
   value: "duckdb"
 });
 
+async function registerExtensions(db) {
+  const connection = await db.connect();
+  try {
+    const extensions = await fetch(import.meta.resolve("observablehq:duckdb_manifest.json")).then((r) => r.json());
```

Comment:
We’ll want to generate code for this on the page, like we do for `registerTable`. When we render the page we’ll know what the desired extensions are, and we can register them so that the `sql` variable awaits their loading.

Reply:
Though now with exported modules, we might also want to consider how a local module could use a default DuckDB client — I guess we need an equivalent JavaScript API for installing and loading self-hosted extensions for when you can’t use the front matter/page initialization.

Reply:
I've implemented a method to allow all the flexibility we want — you have to go through DuckDBClient if you don't want the default LOADs from the configuration, but I think that's fine.

Reply:
The other reason I want to generate the code is that it’ll be a lot faster since you won’t have to wait for a manifest.json to load.

Reply:
yes the manifest is now appropriately baked-in